addons/web/static/src/legacy/js/views/kanban/kanban_controller.js

```diff
@@ -438,7 +438,9 @@ var KanbanController = BasicController.extend({
                 var columnState = self.model.get(column.db_id, {raw: true});
                 var context = columnState.getContext();
                 var state = self.model.get(self.handle, {raw: true});
-                var groupedBy = state.groupedBy[0];
+                // Since a date field can have a granularity in a groupby (date_field:granularity), and viewUtils.getGroupValue requires
+                // only the field name, we have to split out the eventual granularity
+                var groupedBy = (Array.isArray(state.groupedBy[0])) ? state.groupedBy[0][0].split(':')[0] : state.groupedBy[0].split(':')[0];
```

Comment:
I don't understand this line.
`state.groupedBy` is an array of fields (potentially with a granularity), right? So for instance, something like
`['foo', 'date:day', 'bar']`. In kanban, we only consider the first grouping level, so we take `state.groupedBy[0]`, which can thus be either a field, or a field with a granularity. So I would expect 
```js
var groupedBy = state.groupedBy[0].split(':')[0];
```
What am I missing?

Reply:
You are right, I was confused because in kanban_model, parent.groupedBy is an array and was used as a key in view_utils.getGroupValue as if a string, and in kanban_column it is the groupBy field related to the column (so a string), and I never succeeded in getting to this point in a real situation in kanban_controller (but since it is the same format as in the model we know it's an array)

Will clean this up ;p